packages/schematics/angular/utility/clone.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { deepCopy } from '@angular-devkit/core';
+
+
+/**
+ * Copy a source with the supplied properties.
+ * @param source Source of the copy.
+ * @param properties Properties to copy.
+ */
+export function copyWith<T>(source: T, properties: string[]): T {
+  const copy = Object.keys(source)
+    .filter(key => properties.includes(key))
+    .reduce((obj, key) => {
+      // tslint:disable-next-line:no-any
+      (obj as any)[key] = (source as any)[key];
+
+      return obj;
+    }, {} as T);
+
+  return copy;
+}
+
+/**
+ * Copy a source without the supplied properties.
+ * @param source Source of the copy.
+ * @param properties Properties to copy.
+ */
+export function copyWithout<T>(source: T, properties: string[]): T {
```

Comment:
Removing keys is an anti-pattern. I think we should work from a whitelist instead of a blacklist.